import_data_nianiapl.ipynb

```diff
@@ -0,0 +1,291 @@
+   "source": [
+    "pages = 8"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72a8f134",
+   "metadata": {},
+   "source": [
+    "create set to store offer links:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b5f80b64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "links = set()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34a28cf9",
+   "metadata": {},
+   "source": [
+    "create dictionary from which we will create df later:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7ca516b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "slownik = {'date': [], 'adres': [], 'wynagrodzenie': [], 'etat': [], 'autor': [], 'doswiadczenie': [],\n",
+    "           'wiek_niani': [], 'dodatkowe_wymagania': [], 'jezyk1': [], 'jezyk2':[], 'dzieci': [],\n",
+    "           'liczba_dzieci': []}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e962c9a",
+   "metadata": {},
+   "source": [
+    "getting all links and adding offer links:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "06bba98b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for page in range(1, pages+1):\n",
+    "    page_link = f'''https://niania.pl/praca/warszawa?page={page}'''\n",
+    "    response = requests.get(page_link)\n",
+    "    soup = BeautifulSoup(response.text, 'html.parser')\n",
+    "\n",
+    "    tags = soup('a')\n",
+    "\n",
+    "    for tag in tags:\n",
+    "        offer_link = tag.get('href',None)  \n",
+    "        if type(offer_link) == str and offer_link.startswith('https://niania.pl/szukam/'):\n",
+    "            links.add(offer_link)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b42b93b5",
+   "metadata": {},
+   "source": [
+    "how many offers do we have:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "281b8206",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f'number of offers: ', len(links))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3900b71",
+   "metadata": {},
+   "source": [
+    "iterate through each offer to create connection with every page and get necessary data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42457c21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for link in links:\n",
+    "\n",
+    "    response = requests.get(link)\n",
+    "    soup = BeautifulSoup(response.text, 'html.parser')\n",
+    "    soup_str = str(soup)\n",
+    "\n",
+    "    lista = soup.find('ul', class_=\"list-style-none summary\")\n",
+    "    lista_oczekiwania = soup.find_all('li', class_=\"pico-tick-before\")\n",
+    "\n",
+    "    jezyki = soup.find_all('li', class_='pb-3 icon-lang-polish-before')\n",
+    "\n",
+    "    try: #first languge\n",
+    "        jezyk1 = jezyki[0].get_text().strip()\n",
+    "    except Exception:\n",
```

Comment:
jesteś w stanie zamiast Exception napisać jaki to jest exception? np. `except KeyError`? Ogólnie to jest dużo lepsza praktyka, w ten sposób się "kontroluje" błędy, a nie wrzuca każdy błąd do except